src/server/repositories/UrlRepository.ts

```diff
@@ -175,6 +175,28 @@ export class UrlRepository implements UrlRepositoryInterface {
     return this.urlMapper.persistenceToDto(newUrl)
   }
 
+  /**
+   * @param  {string} shortUrl Short url.
+   * @returns {Promise<boolean>} Returns true if shortUrl is available and false otherwise.
+   */
+  public isShortUrlAvailable: (shortUrl: string) => Promise<boolean> = async (
+    shortUrl,
+  ) => {
+    try {
+      // Cache lookup
+      await this.getLongUrlFromCache(shortUrl)
+      // if long url does not exist, throws error
+      // return false if no error since long url exists
+      return false
+    } catch {
+      // Cache failed, look in database
+      const url = await Url.findOne({
+        where: { shortUrl },
+      })
```

Comment:
/nit/micro-optimization: since we only care for an `exists`, you could reduce the selection to just the `shortUrl` field. Otherwise the query seems to be sent like this atm (according to the traces):
```
SELECT shortUrl, longUrl, state, isFile, contactEmail, description, source, tagStrings, createdAt, updatedAt, userId FROM urls WHERE url . shortUrl = ?
```

```suggestion
      const url = await Url.findOne({
        attributes: ['shortUrl'],
        where: { shortUrl },
      })
```